content/Watsonx/WatsonxAI/files/204/use_case_rag.ipynb

```diff
@@ -0,0 +1,313 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Implement RAG Use Cases in watsonx.ai\n",
+    "\n",
+    "In this lab you will review and run examples of LLM applications that implement the Retrieval Augmented Generation (RAG) pattern of working with LLMs. We will expand on the concepts that you learned in the previous labs.\n",
+    "\n",
+    "RAG (Retrieval-Augmented Generation) is one of the most common use cases in generative AI because it allows us to work with data \"external to the model\", for example, data that was not used for model training. Many use cases require working with proprietary company data, and it's one of the reasons why RAG is frequently used in generative AI applications. RAG also allows us to add some guardrails to generated output and reduce hallucination. RAG can be used with several generative AI use cases, including:\n",
+    "\n",
+    "- Question and answer\n",
+    "- Summarization\n",
+    "- Content generation\n",
+    "\n",
+    "> A \"human interaction\" analogy of RAG is providing a document to a person and asking \n",
+    "them to answer question based on the information in the document.\n",
+    "\n",
+    "To get started we'll first verify that you have the necessary dependencies installed to run this notebook.\n",
+    "\n",
+    "Go ahead and run the following code cell."
```

Comment:
Might want to add a note mentioning that this can take a few seconds to complete